packages/experimental/Switch/src/Switch.win32.tsx

```diff
@@ -36,26 +37,80 @@ export const Switch = compose<SwitchType>({
     onOffText: Text,
   },
   useRender: (userProps: SwitchProps, useSlots: UseSlots<SwitchType>) => {
+    const [onOffTextTest, setOnOffTextTest] = React.useState(null);
+    const [finalWidth, setFinalWidth] = React.useState<number>(0);
+    const [textBeingTested, setTextBeingTested] = React.useState<textBeingTestedStates>('init');
+    const { onText, offText } = userProps;
+    const toggleContainerRef = React.useRef(null);
     const switchInfo = useSwitch(userProps);
+
     // grab the styled slots
     const Slots = useSlots(userProps, (layer) => switchLookup(layer, switchInfo.state, switchInfo.props));
 
+    /*
+      Controls the rendering of the onText and offText in order to take measurements of what the minWidth should
+      be. Having the correct minWidth prevents the control from "jiggling" when the onText and offText
+      are different sizes.
+    */
+    const measureOnOffTexts = React.useCallback(
```

Comment:
>const measureOnOffTexts = React.useCallback( [](http://example.com/codeflow?start=4&length=44)

We've generally put all hooks in the use<Component> function. This allows clients to reuse those hooks if they want to use it for a different component or change the render function of the component for some reason.

I think it's fine that the measuring code isn't in the main useSwitch hook but I think it would be good for it to be in its own hook so that if people want to modify the render function, they don't have to rewrite all of this logic.

Reply:
Changed implementation.